File: 941. Valid Mountain Array/ValidMountainArray.cs
public class Solution {
    public bool ValidMountainArray(int[] arr) {

        var (peak, peakIdx) = arr.Select((n, i) => (n, i)).Max();

        if (peakIdx >= arr.Length-1 || peakIdx == 0) return false;

        for (int i = 1; i <= peakIdx; ++i) {
            if (arr[i] <= arr[i-1]) return false;
        }

        for (int i = peakIdx+1; i < arr.Length; ++i) {
            if (arr[i] >= arr[i-1]) return false;
        }

        return true;
    }
}

File: 941. Valid Mountain Array/ValidMountainArray.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        
        peak = max(arr)
        peakIdx = arr.index(peak)

        if peakIdx >= len(arr)-1 or peakIdx == 0:
            return False

        for prevNum, num in zip(arr[:peakIdx],arr[1:peakIdx+1]):
            if num <= prevNum:
                return False

        for prevNum, num in zip(arr[peakIdx:-1],arr[peakIdx+1:]):
            if num >= prevNum:
                return False

        return True
